### PR TITLE
feat: add new field expense in expenseLine form

### PR DESCRIPTION
**Pull request by gca-axelor**
RM#72432

### CODE REVIEW

**Review by vhu-axelor — APPROVED** [review]
Work well, nice! 🤩

File: packages/apps/hr/src/features/expenseLineSlice.js [hunk 1]
```diff
@@ -114,32 +115,39 @@ export const createExpenseLine = createAsyncThunk(
...
+          action:
+            data?.expenseId == null
+              ? 'Hr_SliceAction_FetchExpenseLines'
+              : data?.mode === ExpenseLine.modes.general
+              ? 'Hr_SliceAction_SearchGeneralExpenseLines'
+              : 'Hr_SliceAction_SearchKilometricExpenseLines',
+          getState,
+          data: {expenseId: data.expenseId, userId: data?.userId},
+          responseOptions: {isArrayResponse: true},
+        });
```
Comment (lme-axelor) [inline packages/apps/hr/src/features/expenseLineSlice.js:149]: is _searchGeneralExpenseLines & _searchKilometricExpenseLines still required if you refresh the complete expense ? I think the lines are refreshing themselves when the expense is updated no ?

File: packages/apps/hr/src/components/templates/ExpensePicker/ExpensePicker.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      required={required}
+    />
+  );
+};
+
+const styles = StyleSheet.create({
+  picker: {
+    alignSelf: 'center',
+  },
+});
```
Comment (lme-axelor) [inline packages/apps/hr/src/components/templates/ExpensePicker/ExpensePicker.js:74]: I think this is not required anymore with V improvements on Picker component

File: packages/apps/hr/src/components/templates/ExpensePicker/ExpensePicker.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, {useEffect} from 'react';
+import {StyleSheet} from 'react-native';
+import {Picker} from '@axelor/aos-mobile-ui';
+import {useSelector, useDispatch} from '@axelor/aos-mobile-core';
+import {searchExpenseDraft} from '../../../features/expenseSlice';
+
+const ExpensePickerAux = ({
```
Comment (lme-axelor) [inline packages/apps/hr/src/components/templates/ExpensePicker/ExpensePicker.js:25]: Don't we already have a picker with user's draft expenses ? On the popup to link orphan lines to an expense

**Review by vhu-axelor — CHANGES_REQUESTED** [review]

**Review by vhu-axelor — APPROVED** [review]

File: packages/apps/hr/src/components/templates/ExpensePicker/ExpensePicker.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      emptyValue={false}
+      onValueChange={onChange}
+      required={required}
+      readonly={readonly}
+      isValueItem={true}
+    />
+  );
+};
+
+const ExpensePicker = ({title, defaultValue, onChange, readonly, required}) => {
```
Comment (lme-axelor) [inline packages/apps/hr/src/components/templates/ExpensePicker/ExpensePicker.js:56]: Can you name if DraftExpensePicker please

File: packages/apps/hr/src/features/expenseLineSlice.js [hunk 5]
```diff
@@ -29,7 +29,6 @@ import {
   updateExpenseLine as _updateExpenseLine,
   deleteExpenseLine as _deleteExpenseLine,
 } from '../api/expense-line-api';
-import {ExpenseLine} from '../types';
 import {fetchExpenseById} from './expenseSlice';
 import {updateExpense} from './expenseSlice';
```
Comment (lme-axelor) [inline packages/apps/hr/src/features/expenseLineSlice.js:33]: Can you merge imports please

**Review by vhu-axelor — APPROVED** [review]

**Pull request merged**